### DISCUSSION TITLE
How to replace application_activated and application_deactivated with application_front_switched

### DISCUSSION THREAD

**Discussion by shamil9** [asmvik/yabai#1890] [Q&A] [answered]
Hi, right now I'm using `application_activated` and `application_deactivated` to automatically resize and position and hide when not in focus, my config looks like this

```
yabai -m signal --add event=application_activated \
	app="^Safari$" \
	action="yabai -m rule --add app=\"^Safari$\" manage=off border=off grid=30:4:1:1:2:28"
yabai -m signal --add event=application_deactivated \
	app="^Safari$" \
	action="osascript -e 'tell application \"System Events\" to set visible of process \"Safari\" to false'"
```

How can i achieve same behavior with `application_front_switched`? Thanks

**Comment by asmvik** [comment]
Eh, do you still have this problem? I could reintroduce these two signals without much issue.

**Reply by shamil9** [reply]
That would be nice thank you

**Reply by asmvik** [reply, marked as answer]
Added back on master: https://github.com/koekeishiya/yabai/issues/2122
Works exactly the same way as before.

I noticed that you add a rule in your signal to have perform some action on safari when it gets activated. This rule will be added multiple times (every time the signal runs) and you will end up with an infinite amount in the end. On master you can change it from `rule --add` to `rule --apply` and it will *only apply the rule effects to known windows at the time* instead of storing the rule for future use. See https://github.com/koekeishiya/yabai/issues/2123

Fixed version for the master branch (and next release) would look like:

```
yabai -m signal --add event=application_activated \
	app="^Safari$" \
	action="yabai -m rule --apply app=\"^Safari$\" manage=off grid=30:4:1:1:2:28"
yabai -m signal --add event=application_deactivated \
	app="^Safari$" \
	action="osascript -e 'tell application \"System Events\" to set visible of process \"Safari\" to false'"
```

**Reply by shamil9** [reply]
Thanks, i was not aware of that parameter

**Reply by asmvik** [reply]
I added that parameter just last night, but it will change a bit before it goes into release. You can track this in #2123. The last part here is the one you want, but basically:

 
To re-apply an existing rule to currently known windows in the future:

`yabai -m rule --apply [<RULE_SEL> | <ARGUMENTS>]`

if no argument is given, it will apply all registered rules.
If an index or label is given, that particular rule will apply.

You can also decide to provide the arguments directly, e.g:

`yabai -m rule --apply app="^Safari$" manage=off grid=1:1:0:0:1:1 opacity=1.0`

I've updated my above comment to match this syntax for future references.
